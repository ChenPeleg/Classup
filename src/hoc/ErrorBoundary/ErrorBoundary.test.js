import React from "react";
import { shallow, mount } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

const Something = () => null;
const wrapper = mount(
  <ErrorBoundary>
    <Something />
  </ErrorBoundary>
);
describe('ErrorBoundary', () => {
  beforeEach(() => {
    wrapper.setState({ hasError: false })
  });

  it('should display an ErrorMessage if wrapped component throws', () => {
    const error = new Error('test');
    wrapper.find(Something).simulateError(error);
    expect(wrapper.state().hasError).toBe(true);

  });
  it("should render beacause of error", () => {
    wrapper.setState({ hasError: true });
    expect(wrapper).toMatchSnapshot()
  })
})