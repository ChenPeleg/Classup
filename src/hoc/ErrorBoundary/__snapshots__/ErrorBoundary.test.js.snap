// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render beacause of error 1`] = `ReactWrapper {}`;
